### PR TITLE
feat: add SfDropdown to userIcon

### LINKED ISSUE #643: [default-theme] implement dropdown for user icon
Implementing 

## Acceptance criteria
- [ ] implement SFUI component https://storybook.storefrontui.io/?path=/story/molecules-dropdown--common
- [ ] ability to go to MyAccount page from it
- [ ] ability to logout from it

### PR DESCRIPTION
**Pull request by Michal-Dziedzinski**
## Changes

closes #643 

dropdown

### Checklist

- [x] I followed [contributing](https://github.com/DivanteLtd/shopware-pwa/blob/master/CONTRIBUTING.md) guidelines

### CODE REVIEW

File: packages/default-theme/components/SwBottomNavigation.vue [hunk 1]
```diff
@@ -28,9 +28,24 @@
...
+        class="menu-button"
+      >
+        <template #icon>
+          <SfIcon icon="profile" size="20px" />
+          <SfSelect
+            class="menu-button__select"
+          >
+            <SfSelectOption :value="getPageAccount">
+              <nuxt-link  class="sf-header__link" :to="getPageAccount">
+                Go to my account
```
Comment (patzick) [inline packages/default-theme/components/SwBottomNavigation.vue:40]:
```suggestion
                My account
```

File: packages/default-theme/components/SwBottomNavigation.vue [hunk 2]
```diff
@@ -28,9 +28,24 @@
...
+              </nuxt-link>
+            </SfSelectOption>
+            <SfSelectOption @click.native="logoutUser" :value="'logout'">
```
Comment (patzick) [inline packages/default-theme/components/SwBottomNavigation.vue:43]:
```suggestion
            <!-- TODO: change .native to @click after https://github.com/DivanteLtd/storefront-ui/issues/1097 -->
            <SfSelectOption @click.native="logoutUser" :value="'logout'">
```

**Comment by akrajna** [comment]
* the icon 'My account' is in wrong position as shown below: 
DropDown
* on mobile view when user is not logged, after pressing 'my account' icon there shouldn't be buttons: 'My account' and 'Logout'. Instead the login window should popup. 
DropDownMobile

**Comment by akrajna** [comment]
* on mobile devices, when user is logged in, after pressing the 'My account' icon and selecting "my account' button user should be redirected to my account page and the buttons should be not visible anymore. 
Actual behavior: after pressing 'My account' user is redirected to my account page but buttons: 'My account' and "logout' are visible.
Expected behavior: after pressing 'My account' user is redirected to my account page and buttons: 'My account' and 'logout' are not visible.
DropdownLoggedin

**Comment by akrajna** [comment]
* after logout user is redirected to login window. User shouldn't be redirected to login window after logout. User should be informed about being logout maybe by popup  with 'You are logged out'.

**Comment by Michal-Dziedzinski** [comment]
@akrajna can you create feature issue for that? :) or check that similar exists? I mean popup. Mobile view fixed, Icons position fixed in #694

**Comment by Michal-Dziedzinski** [comment]
blocked by https://github.com/DivanteLtd/storefront-ui/issues/1039

File: packages/default-theme/components/SwTopNavigation.vue [hunk 3]
```diff
@@ -163,13 +184,28 @@ export default {
...
   },
+  computed: {
+    getPageAccount() {
+      return this.$i18n.path(PAGE_ACCOUNT)
+    }
+  },
+  watch:{
+    $route (to, from){
+        this.isDropdownOpen = false;
+    }
```
Comment (patzick) [inline packages/default-theme/components/SwTopNavigation.vue:198]: this should be covered by click-outside directive

**Comment by akrajna** [comment]
On demo instance: 
* the icon 'My account' is in wrong position - the spaces between the icons are too small, as shown below: 
Myaccount1
* after log out the dropdown does not disappear if user clicks outside the dropdown. The dropdown should disappear after clicking outside. 
Logout1

**Comment by akrajna** [comment]
On demo instance:
* after log out the dropdown does not disappear if user clicks outside the dropdown. The dropdown should disappear after clicking outside. 
Logout1

**Comment by akrajna** [comment]
On demo instance
* dropdown should be centrally under the 'My account' icon as shown below:
dropdownCanary
Now it look as below:
dropdownDemo

**Comment by akrajna** [comment]
On demo and canary instance
* on mobile devices, logout button in the popup should be centered
LogoutMobile

**Comment by akrajna** [comment]
On demo instance:
* on mobile devices after pressing log out the dropdown disappears and appears for a moment to disappear again, as shown below:
Logout2
the dropdown should just disappears

**Comment by akrajna** [comment]
On canary instance:
* on mobile view for not logged in user after pressing the 'My account' icon nothing happens, as below:
MyAccountCanary

Expected behavior: for not logged in user after pressing the 'My account' icon   the login window should appears.

**Comment by akrajna** [comment]
The issue for popup with 'You are logged out'  message was created here https://github.com/DivanteLtd/shopware-pwa/issues/763

**Comment by akrajna** [comment]
New bug report is created here #769

**Pull request merged**